### PR TITLE
[FLINK-30619][tests] Add retry

### PR DESCRIPTION
**Pull request by zentol**

### CODE REVIEW

**Comment by XComp** [comment]
What speaks against replacing the `Thread.sleep(10L)` with a waiting condition in the test?
```
CommonTestUtils.waitUtil(
                () -> upTimeGauge.getValue() > 0L,
                TestingUtils.infiniteDuration(),
                "No non-zero uptime was measured.");
```

**Comment by zentol** [comment]
We don't know the root cause, but know that the test is quick and the failure rate is low.
A retry should guarantee that this test will no longer fail; not sure if the same can be said about the loop? LMKWYT.

**Comment by XComp** [comment]
Waiting for the `upTimeGauge` value feels to me like the better approach because we might get the chance of analyzing the thread dump if it really times out forever. :thinking: 
On another note: Did you consider that `taskManagerGateway.waitForSubmissions` is just exceeding its timeout of 5 seconds? `waitForSubmissions` uses `poll` internally which just returns `null` rather than failing in case of reaching the timeout.

**Comment by zentol** [comment]
I doubt that a thread dump is gonna help us much since we know that the job state was changed from the existing logs.
But in the end waiting for it is a de-facto retry so why not.

File: flink-runtime/src/test/java/org/apache/flink/runtime/scheduler/adaptive/AdaptiveSchedulerTest.java [hunk 1]
```diff
@@ -585,7 +586,7 @@ public void testStatusMetrics() throws Exception {
         // sleep a bit to ensure uptime is > 0
         Thread.sleep(10L);
 
-        assertThat(upTimeGauge.getValue()).isGreaterThan(0L);
+        CommonTestUtils.waitUntilCondition(() -> upTimeGauge.getValue() > 0L);
```
Comment (XComp) [inline flink-runtime/src/test/java/org/apache/flink/runtime/scheduler/adaptive/AdaptiveSchedulerTest.java:589]: The `Thread.sleep(10L)` can be removed if we add the waitUntil condition. Additionally, the same pattern with `Thread.sleep` and `isGreatherThan` is applied a few lines further down in the test code. We might want to update that as well.

**Review by XComp — CHANGES_REQUESTED** [review]

**Review by XComp — APPROVED** [review]
LGTM :+1:

**Pull request merged**